Build Cloud Account Id Artefacts 3.0.yml

```diff
@@ -0,0 +1,288 @@
+    inputs:
+      targetType: inline
+      script: >-
+        Write-Host "Writing the Full Image URL $(BuildImage.amiID) in the $(Build.ArtifactStagingDirectory)" | Out-Default | Write-Host
+
+        Out-File -FilePath (Join-Path $(Build.ArtifactStagingDirectory) "$(System.JobDisplayName).txt") -InputObject $(BuildImage.amiID) | Out-Default | Write-Host
+
+        Write-Host "Full Image URL written successfully" | Out-Default | Write-Host
+  - task: AmazonWebServices.aws-vsts-tools.AWSPowerShellModuleScript.AWSPowerShellModuleScript@1
+    name: Vmtest
+    displayName: VM Tests
+    env:
+        TestImageName: $(BuildImage.amiID)
+        TestCloudName: AWS
+        AtomicBuild: $(atomic-build)
+        TestVmName: Test$(System.JobDisplayName)
+        SG: $(System.JobDisplayName)
+        keypair: $(KeypairName)
+        keypairpath: $(keypair.secureFilePath)
+        VersionText: $(System.JobDisplayName)
+    inputs:
+      awsCredentials: $(awsCredentials)
+      regionName: $(regionName)
+      filePath: scripts/invoke-pester-tests.ps1
+      inlineScript: >2-
+
+        # Invoke Pester Tests
+
+        cd $(System.DefaultWorkingDirectory)\Tests
+
+        $result = Invoke-Pester -Script '.\Image*' -OutputFormat  NUnitXml -OutputFile '$(System.DefaultWorkingDirectory)\Test-Vm.xml' -PassThru
+
+        $result | Out-Default | Write-Host
+
+        if ($result.Result -eq "Failed") {
+            throw "Failed Tests Count: $($result.FailedCount)"
+        } else {
+            Write-Host "Tested the image successfully."
+        }
+  - task: PublishTestResults@2
+    displayName: Publish Test Results **/Test-*.xml
+    inputs:
+      testRunner: NUnit
+      testResultsFiles: '**/Test-*.xml'
+      testRunTitle: $(System.JobDisplayName)
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifact: Image URL'
+    inputs:
+      PathtoPublish: $(Build.ArtifactStagingDirectory)/$(System.JobDisplayName).txt
+      ArtifactName: aws
+      FileCopyOptions: ''
+  - task: AmazonWebServices.aws-vsts-tools.AWSPowerShellModuleScript.AWSPowerShellModuleScript@1
+    displayName: Deleting Security group and Instance
+    inputs:
+      awsCredentials: $(awsCredentials)
+      regionName: $(regionName)
+      arguments: -version "$(System.JobDisplayName)"
+      filePath: scripts/terminate-instance.ps1
+      inlineScript: >+
+        # Terminating the Instance
+
+        if('$(TerminateInstance)' -eq 'True') {
+           Write-Host "Removing the instance $(BuildImage.InstanceID) "
+            Remove-EC2Instance -InstanceId $(BuildImage.InstanceID) -Force
+        }
+
+        #Removing Vm
+
+        Write-Host "Removing the Vm"
+
+        Remove-EC2Instance -InstanceId  $(Vmtest.instanceID) -Force
+
+
+        # Deleting the Security Group
+
+        Write-Host "Deleting the security group"
+
+        Start-Sleep -Seconds 180
+         Remove-EC2SecurityGroup -GroupName 'w19d-15-0-$(VersionText-w19d-15-0)' -Force
+
+    
+
+...

```

Comment:
Do these 3 dots "..." mean something?

Reply:
that came from vscode when we were working it. removed it